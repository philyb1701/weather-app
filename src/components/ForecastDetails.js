import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date" data-testid="forecast-details">
        {formattedDate}
      </div>
      <div className="forecast-details__max-temp" data-testid="forecast-temp">
        High: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__max-temp" data-testid="forecast-temp">
        Min: {temperature.min}&deg;C
      </div>
      <div
        className="forecast-details__humidity"
        data-testid="forecast-humidity"
      >
        Humidity: {humidity}%
      </div>
      <div className="forecast-details__wind" data-testid="forecast-wind">
        Wind: {wind.speed}mph
      </div>
      <div
        className="forecast-details__direction"
        data-testid="forecast-direction"
      >
        Direction: {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      direction: PropTypes.string,
      speed: PropTypes.number,
    }),
  }).isRequired,
};

export default ForecastDetails;
