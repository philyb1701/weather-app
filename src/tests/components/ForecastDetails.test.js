import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    temperature: {
      max: 22,
      min: 12,
    },
    humidity: 60,
    wind: {
      direction: "NE",
      speed: 10,
    },
    onForecastSelect() {
      return 1;
    },
  };

  describe("ForecastDetails-snapshot", () => {
    it("renders props correctly", () => {
      const { asFragment } = render(<ForecastDetails forecast={validProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("ForecastDetails-values", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText("Thu Jan 01 1970")).toHaveAttribute(
      "class",
      "forecast-details__date"
    );
    expect(getByText("Max: 22°C")).toHaveAttribute(
      "Class",
      "forecast-details__max-temp"
    );
    expect(getByText("Min: 12°C")).toHaveAttribute(
      "Class",
      "forecast-details__min-temp"
    );
    expect(getByText("Humidity: 60%")).toHaveAttribute(
      "Class",
      "forecast-details__humidity"
    );
    expect(getByText("Wind: 10mph")).toHaveAttribute(
      "Class",
      "forecast-details__wind"
    );
    expect(getByText("Wind: 10mph")).toHaveAttribute(
      "Class",
      "forecast-details__wind"
    );
    expect(getByText("Direction: NE")).toHaveAttribute(
      "Class",
      "forecast-details__direction"
    );
  });
});
