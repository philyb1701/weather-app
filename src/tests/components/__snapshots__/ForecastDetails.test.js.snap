// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails ForecastDetails-snapshot renders props correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
      data-testid="forecast-details"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__max-temp"
      data-testid="forecast-temp"
    >
      Max: 22°C
    </div>
    <div
      class="forecast-details__min-temp"
      data-testid="forecast-temp"
    >
      Min: 12°C
    </div>
    <div
      class="forecast-details__humidity"
      data-testid="forecast-humidity"
    >
      Humidity: 60%
    </div>
    <div
      class="forecast-details__wind"
      data-testid="forecast-wind"
    >
      Wind: 10mph
    </div>
    <div
      class="forecast-details__direction"
      data-testid="forecast-direction"
    >
      Direction: NE
    </div>
  </div>
</DocumentFragment>
`;
